packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/licence-summary.spec.js

```diff
@@ -220,4 +238,70 @@ describe('The licence summary page', () => {
+    expect(data2.statusCode).toBe(302)
+    expect(data2.headers.location).toBe(LICENCE_SUMMARY.uri)
+  })
+
+  it('date of birth (senior) amendment causes redirect the licence summary page and then the contact summary page', async () => {
+    await injectWithCookies('POST', DATE_OF_BIRTH.uri, dobHelper(dobSeniorToday))
+    const data = await injectWithCookies('GET', CONTROLLER.uri)
+    expect(data.statusCode).toBe(302)
+    expect(data.headers.location).toBe(LICENCE_SUMMARY.uri)
+    await injectWithCookies('GET', LICENCE_SUMMARY.uri)
+    await injectWithCookies('POST', LICENCE_SUMMARY.uri)
+
+    const data2 = await injectWithCookies('GET', CONTROLLER.uri)
+    expect(data2.statusCode).toBe(302)
+    expect(data2.headers.location).toBe(NAME.uri)
+  })
+
+  it('date of birth amendment (no licence required) causes a redirect to the no licence required page', async () => {
+    await postRedirectGet(LICENCE_TO_START.uri, { 'licence-to-start': 'after-payment' })
+    const data = await postRedirectGet(DATE_OF_BIRTH.uri, dobHelper(dobJuniorTomorrow))
+    expect(data.statusCode).toBe(302)
+    expect(data.headers.location).toBe(NO_LICENCE_REQUIRED.uri)
+    await injectWithCookies('GET', LICENCE_SUMMARY.uri)
+    await injectWithCookies('POST', LICENCE_SUMMARY.uri)
+
+    const data2 = await injectWithCookies('GET', CONTROLLER.uri)
+    expect(data2.statusCode).toBe(302)
+    expect(data2.headers.location).toBe(NAME.uri)
+  })
+
+  it('date of birth amendment (junior) causes redirect to the junior licence page and subsequent redirect to the licence summary page and then the contact summary page', async () => {
+    await injectWithCookies('POST', DATE_OF_BIRTH.uri, dobHelper(dobJuniorToday))
+    let data = await injectWithCookies('GET', CONTROLLER.uri)
+    expect(data.statusCode).toBe(302)
+    expect(data.headers.location).toBe(JUNIOR_LICENCE.uri)
+    data = await postRedirectGet(JUNIOR_LICENCE.uri, {})
+    expect(data.statusCode).toBe(302)
+    expect(data.headers.location).toBe(LICENCE_SUMMARY.uri)
+
+    await injectWithCookies('GET', LICENCE_SUMMARY.uri)
+    await injectWithCookies('POST', LICENCE_SUMMARY.uri)
+
+    const data2 = await injectWithCookies('GET', CONTROLLER.uri)
+    expect(data2.statusCode).toBe(302)
+    expect(data2.headers.location).toBe(NAME.uri)
+  })
+
+  it('date of birth amendment (junior) causes a method of contact of letter to be set no none', async () => {
+    await injectWithCookies('POST', DATE_OF_BIRTH.uri, dobHelper(dobSeniorToday))
+    await injectWithCookies('GET', CONTROLLER.uri)
+    await postRedirectGet(CONTACT.uri, { 'how-contacted': 'none' })
+    const { payload } = await injectWithCookies('GET', TEST_TRANSACTION.uri)
+    expect(JSON.parse(payload).permissions[0].licensee.preferredMethodOfConfirmation).toEqual(HOW_CONTACTED.letter)
+    expect(JSON.parse(payload).permissions[0].licensee.preferredMethodOfReminder).toEqual(HOW_CONTACTED.letter)
```

Comment:
This seems to diverge from the stated ambition of the test - we're testing for a letter contact? I think this should be a separate test

Reply:
I am not sure I understand. The juniors cannot get a letter. This is stated in the test description: 'date of birth amendment (junior) causes a method of contact of letter to be set no none'

Reply:
I just meant that, from the specification of the test, we really just need to test that the preferredMethodOfConfirmation and preferredMethodOfReminder are both 'none' , checking that they're 'letter' as a default is beyond the scope of this test. If you want to test it, it should be in a separate test, I think

Reply:
ok